util/types/binary_literal.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+)
+
+// BinaryLiteral is the internal type for storing bit / hex literal type.
+type BinaryLiteral struct {
+	// Value holds the raw value for literal in BigEndian.
+	Value []byte
+}
+
+// BitLiteral is the bit literal type.
+type BitLiteral struct {
+	BinaryLiteral
+}
+
+// HexLiteral is the hex literal type.
+type HexLiteral struct {
```

Comment:
It seems like `HexLiteral` is not actually used, can we just remove this type?

Reply:
BitLiteral and HexLiteral is only used in parser, to distinguish the literal type in `DefaultTypeForValue@field_type.go` (because when using them to build fields, the field has different flags).

Reply:
We can just use BinaryLiteral directly in parser to replace `HexLiteral`.